src/lib/hooks/useConnectionGun.js

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { AppState } from 'react-native'
+import Config from '../../config/config'
+import logger from '../logger/pino-logger'
+import SimpleStore from '../undux/SimpleStore'
+
+let isFirstCheckGun = false
+let needToBindEventsGun = true
+
+const log = logger.child({ from: 'useHasConnectionGun' })
+
+export default () => {
+  const [isConnection, setIsConnection] = useState(true)
+  const store = SimpleStore.useStore()
+  const userStorage = store.get('userStorage')
+  const connectionCheck = useRef()
+
+  const killLastConnectionCheck = () => {
+    if (typeof connectionCheck.current === 'function') {
```

Comment:
please add comments through out this file regarding the new logic implemented here, how it works. and comments above relevant line codes